### PR TITLE
Check for Amazon in osfamilymap to create pkg repo

### LINKED ISSUE #65: [FEATURE] Amazon Linux 2 pkg repo support
https://github.com/saltstack-formulas/prometheus-formula/blob/master/prometheus/osfamilymap.yaml doesn't take into account the fact that Amazon Linux 2 resolves `$releasever` to `2`, instead of 7 or 8, so the yum repository definition isn't rendered to a valid el/7 or el/8 packagecloud URL.  Low-pri, but thought I'd bring it up.

### PR DESCRIPTION
**Pull request by mdschmitt**
Fixes #65

### CODE REVIEW

File: prometheus/osfamilymap.yaml [hunk 1]
```diff
@@ -443,3 +448,7 @@ MacOS:
         archive:
           source_hash: 15132494523c2b6a89e09b2da63452c8fe587fb82fcc3fd21cc75a4aa2766644
 
+
```
Comment (noelmcloughlin) [inline prometheus/osfamilymap.yaml:451]: remove the extra lines..

File: prometheus/osfamilymap.yaml [hunk 2]
```diff
@@ -206,9 +206,14 @@ RedHat:
           args:
             config.file: /etc/prometheus/blackbox.yml
     repo:
+      {%- if grains.os == 'Amazon' %}
+        {%- set releasever = salt['cmd.run']("rpm -E '%{rhel}'") %}
```
Comment (noelmcloughlin) [inline prometheus/osfamilymap.yaml:210]: Indent 4 spaces, instead of 2 spaces, to line up with the block above (suggestion, not rule).

**Review by noelmcloughlin — CHANGES_REQUESTED** [review]
LGTM, two suggested updates

**Comment by mdschmitt** [comment]
Updated.  Not sure how those extra lines ended up in there.  Thx for pointing that and the spacing out.  Should be all good now.

**Review by myii — CHANGES_REQUESTED** [review]
This would be so much neater if there was an `osmap.yaml` available but that's not present unfortunately.

The testing should be updated though:

https://github.com/saltstack-formulas/prometheus-formula/blob/2c572d0019fbbac51e2a5cf70b8928496be2cae9/.gitlab-ci.yml#L170-L172

* Please comment out line 170 and then uncomment line 172, which will test both the `default` and `repo` suites for Amazon.

**Comment by mdschmitt** [comment]
is `tar` ...not installed on Amazon Linux by default?  That makes no sense.

**Comment by mdschmitt** [comment]
well, for the archive install, it's supposed to verify that..  I'm confused.

**Comment by myii** [comment]
> is `tar` ...not installed on Amazon Linux by default? That makes no sense.

@mdschmitt This will resolve your problem.  Change the first line of this block:

https://github.com/saltstack-formulas/prometheus-formula/blob/2c572d0019fbbac51e2a5cf70b8928496be2cae9/prometheus/clientlibs/install.sls#L31-L36

To:

```diff
-        {%- if grains.get('osfinger', '') in ['Oracle Linux Server-8'] %}
+        {%- if grains.get('osfinger', '') in ['Oracle Linux Server-8', 'Amazon Linux-2'] %}
```

**Comment by mdschmitt** [comment]
I'll admit, I could've found that out eventually, but I really appreciate your help.  Updated and pushed.  Thank you!

**Comment by mdschmitt** [comment]
...and pipeline success!

**Review by myii — APPROVED** [review]

**Comment by myii** [comment]
> I'll admit, I could've found that out eventually, but I really appreciate your help. Updated and pushed. Thank you!

@mdschmitt You're welcome.  And thank you for your excellent contribution!

**Pull request merged**